Eradication/Assortivity.cpp

```diff
@@ -350,29 +350,54 @@ namespace Kernel
         return group ;
     }
 
-    struct PartnerScore
+    IIndividualHumanSTI* Assortivity::SelectPartnerFromScore( float total_score,
+                                                              const std::vector<float>& rScores,
+                                                              const std::vector<IIndividualHumanSTI*>& rPartners )
     {
-        IIndividualHumanSTI* pPartner ;
-        float score ;
+        // -------------------------------------------------------------------------
+        // --- Select the partner based on their score/probability of being selected
+        // -------------------------------------------------------------------------
+        release_assert(m_pRNG != nullptr);
+        float ran_score = m_pRNG->e() * total_score;
+        std::vector<float>::const_iterator it = std::lower_bound( rScores.begin(),
+                                                                  rScores.end(),
+                                                                  ran_score );
+        if( it != rScores.end() )
+        {
+            // -------------------------------------------------------------------
+            // --- The old code was strictly greater than so this find this finds
+            // --- the next value that is greater than.  This could go away if we
+            // --- want to accept this small change in partner selection.
+            // -------------------------------------------------------------------
+            while( (it != rScores.end()) && (*it == ran_score) )
+            {
```

Comment:
is it not very unlikely that ( "random number between 0 and 1" * "total_score") be exactly equal to a float we're pointing to right now? seems like it would almost always to the point of just always be false?  because floats are.. floaty?

Reply:
and so we would then just use the it.begin() in the if underneath? and the index would be.. 0?

Reply:
OKAY! I get it now. Thank you @kfrey-idm  and @Bridenbecker  for explaining. This is handling the edge case if the random number we pick just HAPPENS to be the actual lower bound edge because lower_bound() does "less than" and we are taking care of here in case this is actually equals to the next lower bound and therefore should be picking the next person.